FRAuth/FRAuth/DeviceBinding/ApplicationPinDeviceAuthenticator.swift

```diff
@@ -0,0 +1,63 @@
+// 
+//  ApplicationPinDeviceAuthenticator.swift
+//  FRAuth
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+import LocalAuthentication
+import FRCore
+
+
+/// DeviceAuthenticator adoption for Application Pin authentication
+open class ApplicationPinDeviceAuthenticator: DeviceAuthenticator, CryptoAware {
```

Comment:
Why is this open and not public? Are we offering the users to make their own versions of it? Is there an example where this is needed? Pardon my ignorance on the subject here if this has been discussed before

Reply:
@george-bafaloukas-forgerock it's open on purpose as we are allowing  them to extend. @jeyanthanperiyasamy can explain the reason and the uses better

Reply:
yeah @george-bafaloukas-forgerock  , there is a usecase like someone can extend  ApplicationPinDeviceAuthenticator  and override the functionalities and inject as well

Example:

class CustomApplicationPinAuthenticator: ApplicationPinDeviceAuthenticator {}

class CustomDeviceBindingCallback: DeviceBindingCallback {
 override fun type() -> DeviceBindingAuthenticationType {
   return CustomApplicationPinAuthenticator()
}

}